packaging/packager_deb.sh

```diff
@@ -5,46 +5,61 @@ readonly distro="${DISTRO:-ubuntu}"
 readonly distro_version="${VERSION:-xenial}"
 readonly distro_root="${ROOT:-$HOME}"
 
-# deb dependencies
+echo "------------- deb dependencies -------------"
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
-# python3
-apt-get -y install python3 python3-coverage
-# python-imaging
-apt-get -y install python3-pil
-# pip dependencies (for dependencies not available as DEB)
+echo "------------- python3 install -------------"
+apt-get -y install python3 python3-venv
+echo "----- pip dependencies (for dependencies not available as DEB) -----"
 apt-get -y install gcc libx11-dev libxtst-dev python3-dev libpng-dev python3-pip
+
+echo "----- create a venv to use that as the python env (needed for python 3.12 compatibility specific to Ubuntu) -----"
+python3 -m venv docker_venv --system-site-packages
+source docker_venv/bin/activate
+
+echo "------------- upgrade pip -------------"
 pip3 install --upgrade pip
-# contour, template, feature, cascade, text matching
+
+echo "------------ install coverage ------------"
+pip3 install coverage
+echo "------------- python-imaging -------------"
+apt-get -y install python3-pil
+
+echo "------------- contour, template, feature, cascade and matching -------------"
 apt-get -y install python3-numpy
 if [[ $distro_version == "xenial" ]]; then
     export DISABLE_OPENCV=1
 else
     apt-get -y install python3-opencv
 fi
-# text matching
+
+echo "------------- text matching -------------"
 if [[ $distro_version == "xenial" ]]; then
     export DISABLE_OCR=1
 else
     apt-get -y install tesseract-ocr libtesseract-dev
     apt-get -y install g++ pkg-config
-    pip3 install pytesseract==0.3.4 tesserocr==2.5.1
+    pip3 install pytesseract==0.3.13 tesserocr==2.7.1
+fi
+
+echo "------------- deep learning -------------"
+pip3 install torch==2.2.0 torchvision==0.17.0
+
+echo "------------- display controlling -------------"
+if [[ -n "$DISABLE_AUTOPY" && "$DISABLE_AUTOPY" == "1" ]]; then
+  export DISABLE_AUTOPY=1
+else
+  pip3 install autopy
 fi
-# deep learning
-pip3 install torch==1.8.1 torchvision==0.9.1
-# screen controlling
-pip3 install autopy==4.0.0
-# TODO: vncdotool doesn't control its Twisted which doesn't control its "incremental" dependency
-pip3 install incremental==22.10.0
 pip3 install vncdotool==0.12.0
 apt-get -y install xdotool x11-apps imagemagick
-apt-get -y install python3-tk scrot
-# TODO: install PyScreeze separately to replace the one PyAutoGUI will install which is incompatible with the current Pillow version 
-pip3 install pyscreeze==0.1.28
-pip3 install pyautogui==0.9.53
+# NOTE: You must install tkinter to use MouseInfo
```

Comment:
Is tkinter automatically installed on Fedora? Isn't MouseInfo needed there?

Reply:
Seems like that, some package may bring it there but I had a run with issues if I don't install tk on the deb

Reply:
So it has been confirmed it is auto-installed on fedora in the CI? If so it can be confirmed it is not implicitly installed on Ubuntu?